core/domain/state_domain.py

```diff
@@ -0,0 +1,543 @@
+# coding: utf-8
```

Comment:
Shouldn't there be tests for this file?

Reply:
Yep, I'll push tests for this tomorrow. Thanks!

Reply:
Moved relevant tests from exp_domain_test to state_domain_test. Can you PTAL @seanlip? Thanks!